#### Companies invest heavily in marketing campaigns across multiple platforms but lack clear insights into which campaigns, channels, and strategies generate the highest return on investment (ROI).
#### This project analyzes marketing performance across Nykaa, Purplle, and Tira to optimize campaign effectiveness and budget allocation.

In [1]:
import pandas as pd
import numpy as np

In [19]:
nykaa = pd.read_csv('../data/nykaa_campaign_data.csv')
purplle = pd.read_csv('../data/purplle_campaign_data.csv')
tira = pd.read_csv('../data/tira_campaign_data.csv')

In [4]:
nykaa['Platform'] = 'Nykaa'
purplle['Platform'] = 'Purplle'
tira['Platform'] = 'Tira'

In [5]:
df = pd.concat([nykaa, purplle, tira], ignore_index=True)

In [6]:
df.head()
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166665 entries, 0 to 166664
Data columns (total 17 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Campaign_ID       166665 non-null  object 
 1   Campaign_Type     166665 non-null  object 
 2   Target_Audience   166665 non-null  object 
 3   Duration          166665 non-null  int64  
 4   Channel_Used      166665 non-null  object 
 5   Impressions       166665 non-null  int64  
 6   Clicks            166665 non-null  int64  
 7   Leads             166665 non-null  int64  
 8   Conversions       166665 non-null  int64  
 9   Revenue           166665 non-null  int64  
 10  Acquisition_Cost  166665 non-null  float64
 11  ROI               166665 non-null  float64
 12  Language          166665 non-null  object 
 13  Engagement_Score  166665 non-null  float64
 14  Customer_Segment  166665 non-null  object 
 15  Date              166665 non-null  object 
 16  Platform          16

,Duration,Impressions,Clicks,Leads,Conversions,Revenue,Acquisition_Cost,ROI,Engagement_Score
count,166665.000000,166665.000000,166665.000000,166665.000000,166665.000000,1.666650e+05,166665.000000,166665.000000,166665.000000
mean,17.490367,55060.854091,4682.367546,1871.514181,1029.088885,5.139066e+05,376.094881,2.690567,13.766293
std,7.504843,25970.653994,3176.502340,1429.590998,858.011218,4.876443e+05,534.939555,4.481216,6.330005
min,5.000000,10001.000000,202.000000,48.000000,17.000000,3.895000e+03,8.180000,-0.990000,2.560000
25%,11.000000,32566.000000,2109.000000,779.000000,401.000000,1.776610e+05,106.730000,0.040000,8.380000
50%,17.000000,55110.000000,3904.000000,1476.000000,776.000000,3.592500e+05,208.510000,1.230000,13.590000
75%,24.000000,77574.000000,6688.000000,2598.000000,1404.000000,6.847970e+05,427.570000,3.580000,18.790000
max,30.000000,100000.000000,14944.000000,8876.000000,6686.000000,4.579910e+06,15473.160000,79.300000,30.990000


In [7]:
df.isnull().sum()
df.drop_duplicates(inplace=True)

In [8]:
df['Date'] = pd.to_datetime(df['Date'])

C:\Users\RENTKAR\AppData\Local\Temp\ipykernel_7340\2394721818.py:1: UserWarning: Parsing dates in %d-%m-%Y format when dayfirst=False (the default) was specified. Pass `dayfirst=True` or specify a format to silence this warning.
  df['Date'] = pd.to_datetime(df['Date'])


In [9]:
df['CTR'] = df['Clicks'] / df['Impressions']
df['Conversion_Rate'] = df['Conversions'] / df['Clicks']
df['Cost_per_Conversion'] = df['Acquisition_Cost'] / df['Conversions']

In [20]:
# Platform Comparison

df.groupby('Platform')['Revenue'].sum()

Platform
Nykaa      28656364282
Purplle    28540346627
Tira       28453535162
Name: Revenue, dtype: int64

#### Insights
- Nykaa generates the highest total revenue (28.66 billion), followed closely by Purplle (28.54 billion) and Tira (28.45 billion).

#### Business Impact
- Prioritize budget allocation towards Nykaa platform campaigns to leverage its revenue-generating potential and optimize overall marketing ROI.

In [21]:
# Channel Performance

df.groupby('Channel_Used')['Conversions'].sum()

Channel_Used
Email                           9782981
Email, Facebook                 1951911
Email, Facebook, Google          440091
Email, Facebook, Instagram       437244
Email, Facebook, WhatsApp        483505
                                 ...   
YouTube, WhatsApp               1912543
YouTube, WhatsApp, Email         477069
YouTube, WhatsApp, Facebook      484903
YouTube, WhatsApp, Google        453300
YouTube, WhatsApp, Instagram     419135
Name: Conversions, Length: 156, dtype: int64

#### Insights
- Email channels show the highest total conversions, indicating strong performance in direct communication strategies.

#### Business Impact
- Invest more in email marketing campaigns to drive higher conversion rates and customer acquisition.

In [22]:
# Campaign Type Analysis

df.groupby('Campaign_Type')['ROI'].mean()

Campaign_Type
Email           2.693300
Influencer      2.648540
Paid Ads        2.714927
SEO             2.688255
Social Media    2.708121
Name: ROI, dtype: float64

#### Insights
- Paid Ads have the highest average ROI (2.71), followed closely by Social Media (2.71) and SEO (2.69).

#### Business Impact
- Focus budget on Paid Ads campaigns to maximize return on investment across marketing efforts.

In [23]:
# ROI Analysis

df.sort_values(by='ROI', ascending=False)

,Campaign_ID,Campaign_Type,Target_Audience,Duration,Channel_Used,Impressions,Clicks,Leads,Conversions,Revenue,Acquisition_Cost,ROI,Language,Engagement_Score,Customer_Segment,Date,Platform,CTR,Conversion_Rate,Cost_per_Conversion
156516,TI-CMP-46406,Social Media,Tier 2 City Customers,7,"Google, Instagram",97086,13165,7856,6152,4041864,8.18,79.30,Hindi,27.99,Tier 2 City Customers,2024-10-17,Tira,0.135601,0.467300,0.001330
12007,NY-CMP-13007,Paid Ads,Youth,29,"Email, YouTube, Facebook",99483,14620,8708,6686,4579910,9.08,74.42,Hindi,30.17,Tier 2 City Customers,2025-02-14,Nykaa,0.146960,0.457319,0.001358
128459,TI-CMP-18349,Paid Ads,Youth,28,"Facebook, Google",99420,13436,7698,5358,4045290,10.32,72.16,English,26.65,Tier 2 City Customers,2025-02-19,Tira,0.135144,0.398779,0.001926
96528,PU-CMP-41973,Paid Ads,Working Women,8,Instagram,85967,12210,6955,5114,3467292,9.99,66.84,Hindi,28.24,Tier 2 City Customers,2024-09-26,Purplle,0.142031,0.418837,0.001953
21689,NY-CMP-22689,Influencer,Youth,20,"Instagram, YouTube",97281,13710,7491,5315,3821485,10.68,66.32,Hindi,27.26,College Students,2025-01-11,Nykaa,0.140932,0.387673,0.002009
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
111469,TI-CMP-1359,Influencer,Working Women,13,"WhatsApp, Facebook, Instagram",12568,330,75,36,9180,8310.08,-0.97,Tamil,3.51,Tier 2 City Customers,2025-05-12,Tira,0.026257,0.109091,230.835556
23406,NY-CMP-24406,Paid Ads,Premium Shoppers,5,"Facebook, YouTube",11625,269,90,27,6183,6805.56,-0.97,Tamil,3.32,Youth,2024-11-20,Nykaa,0.023140,0.100372,252.057778
8361,NY-CMP-9361,Social Media,Tier 2 City Customers,12,"WhatsApp, Email",12784,271,58,32,7584,7187.34,-0.97,English,2.82,College Students,2025-02-10,Nykaa,0.021198,0.118081,224.604375
130848,TI-CMP-20738,Influencer,Premium Shoppers,6,"Instagram, WhatsApp, Email",11146,316,71,22,6072,12723.55,-0.98,Tamil,3.67,Tier 2 City Customers,2025-02-08,Tira,0.028351,0.069620,578.343182


#### Insights
- Sorting campaigns by ROI descending reveals the top-performing campaigns in terms of return on investment.

#### Business Impact
- Prioritize and replicate strategies from high-ROI campaigns to improve overall marketing efficiency.

In [24]:
# Engagement Impact

df[['Engagement_Score','Conversions']].corr()

,Engagement_Score,Conversions
Engagement_Score,1.000000,0.642426
Conversions,0.642426,1.000000


#### Insights
- There is a positive correlation (0.64) between Engagement Score and Conversions, indicating that higher engagement leads to more conversions.

#### Business Impact
- Focus on strategies that boost engagement scores to increase conversion rates and overall campaign effectiveness.

In [25]:
# Platform Efficiency

df.groupby('Platform')[['CTR','Conversion_Rate']].mean()

,CTR,Conversion_Rate
Platform,,
Nykaa,0.085087,0.219716
Purplle,0.084977,0.219068
Tira,0.085006,0.219578


#### Insights
- All platforms show similar average CTR (around 0.085) and Conversion Rates (around 0.22), with Nykaa having a slight edge.

#### Business Impact
- Since platform efficiency is comparable, choose platforms based on audience reach, cost, and strategic alignment rather than performance metrics alone.

In [26]:
# Cost Analysis

df.groupby('Channel_Used')['Cost_per_Conversion'].mean()

Channel_Used
Email                           2.058295
Email, Facebook                 2.013748
Email, Facebook, Google         2.650697
Email, Facebook, Instagram      1.794385
Email, Facebook, WhatsApp       1.723436
                                  ...   
YouTube, WhatsApp               2.055047
YouTube, WhatsApp, Email        1.624218
YouTube, WhatsApp, Facebook     2.755246
YouTube, WhatsApp, Google       1.795054
YouTube, WhatsApp, Instagram    3.099351
Name: Cost_per_Conversion, Length: 156, dtype: float64

#### Insights
- Cost per conversion varies significantly across channel combinations, with some email and social media combinations showing lower costs.

#### Business Impact
- Optimize channel selection by prioritizing cost-effective combinations to reduce acquisition costs while maintaining conversion rates.

In [27]:
df.to_csv("../data/final_data.csv", index=False)